src/node/rpc/node_frontend.h

```diff
@@ -1807,6 +1807,29 @@ namespace ccf
         .set_forwarding_required(endpoints::ForwardingRequired::Never)
         .set_auto_schema<void, nlohmann::json>()
         .install();
+
+      auto process = [this](auto& args, const nlohmann::json& params) {
+        auto node_configuration_subsystem =
+          this->context.get_subsystem<NodeConfigurationSubsystem>();
+        if (!node_configuration_subsystem)
+        {
+          return make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            "NodeConfigurationSubsystem is not available");
+        }
+
+        auto r = nlohmann::json();
+        r["stop_notice"] =
+          node_configuration_subsystem->has_received_stop_notice();
+        return make_success(r);
```

Comment:
Why not fold it in the existing [`GET /node/state`](https://microsoft.github.io/CCF/main/operations/operator_rpc_api.html#get--node-state) endpoint?

Reply:
Good point.